clang/test/SemaTemplate/temp-param-list-dedup.cpp

```diff
@@ -0,0 +1,57 @@
+// RUN: %clang_cc1 %s -verify
```

Comment:
I'd like to also see some testing of the case where the template arguments have different sugar but are canonically the same, and when they differ by only cv-qualifiers. Those seem like likely cases for us to get wrong. Eg:

```c++
using T = int;
using CT = const T;
static_assert(__is_same(__type_list_dedup<TypeList, T, int, const int, const T, CT, const CT>,
                        TypeList<int, const int>));
```

Reply:
Done. I have also added a few tests with arrays and pointers, which should also exhibit a few more combinations of types that subsume qualifiers and could be different when canonical/non-canonical.